src/test/java/UserManagerTest.java

```diff
@@ -0,0 +1,153 @@
+import app.UserManager;
+import app.UserManagerFactory;
+import org.junit.Test;
+
+import javax.servlet.ServletContext;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.HttpSessionContext;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class UserManagerTest {
+    private static HttpSession mock(Class<HttpSession> httpSessionClass) {
```

Comment:
Just a small note that you can use @BeforeClass to run the mock function before all the test are run :) Besides that all good

Reply:
right! it would probably also be good to put the UserManager instance in a @BeforeClass method
ill fix this!!